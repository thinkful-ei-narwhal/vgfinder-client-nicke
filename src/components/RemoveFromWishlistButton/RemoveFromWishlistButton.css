.remove-button {
  margin-top: 15px;
  margin-bottom: 15px;
  color: black;
  background-color: lightgray;
}
