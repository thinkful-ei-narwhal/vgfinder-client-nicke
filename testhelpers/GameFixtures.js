const GameFixtures = {
  stubGameData: {
    id: "1",
    title: "Hollow Knight",
    description:
      "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
    genre: "Adventure",
    rating: "4",
    developer: "Team Cherry",
    trailer_url:
      "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
    image_url_box_art:
      "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
    image_url_two:
      "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
    image_url_three:
      "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
  },
  stubGameListData: [
    {
      id: "1",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
    {
      id: "2",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
  ],
};

export default GameFixtures;
