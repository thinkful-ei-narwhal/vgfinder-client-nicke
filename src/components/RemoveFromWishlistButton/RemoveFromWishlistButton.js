import './RemoveFromWishlistButton.css'
import React, { Component } from 'react'
import UserIdContext from '../../contexts/UserIdContext';
import WishlistService from '../../services/wishlist-api-service'

export default class RemoveFromWishlistButton extends Component {
  //needs to know the user wishlist
  static contextType = UserIdContext;

  removeFromArray(array, arrayItemToRemove) {
    const index = array.indexOf(arrayItemToRemove);
    if (index > -1) {
      array.splice(index, 1);
    }
    return array;
  }

  removeFromWishlist = () => {
    const userWishlist = this.context.wishlist;
    const userGames = this.context.userGames;
    const userId = this.context.userId;
    const game = this.props.game;
    const removeWishlistItem = userWishlist.find(wishList => wishList.game_id === this.props.game.id);

    WishlistService.removeFromWishlist(removeWishlistItem.id)
      .then(res => {
        const updatedWishlist = this.removeFromArray(userWishlist, removeWishlistItem);
        const updatedUserGames = this.removeFromArray(userGames, game);
        this.context.setUserIdWishlistAndGames(userId, updatedWishlist, updatedUserGames);
      })
      .catch(res => {
        this.setState({ error: res.error })
      });
  }

  render() {
    return (
      <button onClick={this.removeFromWishlist}>
        Remove From Wishlist
      </button>
    )
  }
}