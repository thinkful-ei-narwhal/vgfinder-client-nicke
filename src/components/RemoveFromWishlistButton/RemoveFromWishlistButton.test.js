import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RemoveFromWishlistButton from './RemoveFromWishlistButton'
import GameFixtures from './../../../testhelpers/GameFixtures'

describe(`RemoveFromWishlistButton component`, () => {
  it('renders a RemoveFromWishlistButton component with supplied props', () => {

    const wrapper = shallow(<RemoveFromWishlistButton game={GameFixtures.stubGameData} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
