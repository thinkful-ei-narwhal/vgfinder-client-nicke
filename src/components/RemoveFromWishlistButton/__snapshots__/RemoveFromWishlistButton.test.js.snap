// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveFromWishlistButton component renders a RemoveFromWishlistButton component with supplied props 1`] = `
<button
  onClick={[Function]}
>
  Remove From Wishlist
</button>
`;
